addons/l10n_es_edi_sii/models/account_edi_format.py

```diff
@@ -694,3 +706,40 @@ def _l10n_es_edi_sii_post_invoices(self, invoices):
                 })
                 res[inv]['attachment'] = attachment
         return res
+
+    def _l10n_es_edi_sii_cancel_invoices(self, invoices):
+        # Ensure a certificate is available.
+        # TODO: this seems double stuff
+        certificate = invoices.company_id.l10n_es_edi_certificate_id
+        if not certificate:
+            return {inv: {
+                'error': _("Please configure the certificate for SII."),
+                'blocking_level': 'error',
+            } for inv in invoices}
+
+        # Ensure a tax agency is available.
+        l10n_es_edi_tax_agency = invoices.company_id.mapped('l10n_es_edi_tax_agency')[0]
+        if not l10n_es_edi_tax_agency:
+            return {inv: {
+                'error': _("Please specify a tax agency on your company for SII."),
+                'blocking_level': 'error',
+            } for inv in invoices}
+
+        # Generate the JSON.
+        info_list = self._l10n_es_edi_get_invoices_info(invoices)
+
+        # Call the web service.
+        res = self._l10n_es_edi_call_web_service_sign(invoices, info_list, cancel=True)
+
+        for inv in invoices:
+            if res.get(inv, {}).get('success'):
```

Comment:
What happens if there's a failure? I guess the user will have to post the cancellation from the web portal manually, right? Is there some kind of notification, log or anything actionable in that case?

Reply:
It will put a red banner in a similar way as you will have an error when sending.